tests/testsuite/jobserver.rs

```diff
@@ -199,6 +199,7 @@ test-runner:
     p.process(make)
         .env("PATH", path)
         .env("CARGO", cargo_exe())
+        .env("RUSTUP_HOME", env!("REAL_RUSTUP_HOME"))
```

Comment:
Why do we need `REAL_RUSTUP_HOME` rather than just reading `RUSTUP_HOME`?

Reply:
Some of the tests seem to require something in the real rustup installation. I haven't tried to figure out what, though.

Reply:
I can understand forwarding `RUSTUP_HOME`.  What I'm wondering about is `REAL_RUSTUP_HOME`.  We should be able to read `RUSTUP_HOME` (ideally at runtime, not compile time) and forward that to the test.  The `p.process` call is stripping `RUSTUP_HOME` but the process we are currently running in should still have it set *if* it is set (it isn't always)